src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,40 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
+
+  die() {
+    Animal.alive = Animal.alive.filter((animal) => animal !== this);
+  }
```

Comment:
The `die` method uses `filter` to remove the dead animal from the `Animal.alive` array, which is good for avoiding nested loops and aligns with the checklist item regarding code performance. No changes are necessary here.